addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -13,6 +13,31 @@ import { unique } from "@web/core/utils/arrays";
 const qweb = core.qweb;
 const _t = core._t;
 
+const allFormsInfo = new Map();
```

Comment:
Correct me if I'm wrong.
This map inventories form info per `formEl`, so those who want to access it also need to have access to `formEl`.
Can't we just use a `formInfo` field inside `formEl` to do that - which would implicitly take care of the cleanup when the element is garbage collected ? This would also avoid having a static collection which I find scary outside the context of registries.

Reply:
The thing is we have access to the form info in the `WebsiteFormEditor` and we need to access it in the `WebsiteFieldEditor` to compute the domain. So we need a way to communicate between those two widgets. I think that having a static collection is the best way to go. We already do the same kind of thing in the social media option (see `dbSocialValues`) and we clear the collection when we leave the edit mode.

Reply:
It feels different from `dbSocialValues`: in that case the key of the map is a string - so information could not be stored within the string, another structure was needed. Here the key is an actual object with fields, relying on an external structure to maintain a value seems overkill to me.

I.e. what I mean is: why don't we simply
```js
this.$target[0].formInfo = formInfo; // in _renderCustomXML
```
and
```js
formEl.formInfo // both uses in _getDomain
```

Reply:
Even if it could work, we shouldn't rely on this kind of behavior. Adding a property to a dom element is not supported and should not be used AFAIK, because even if it works today, there's no guarantee that it will continue to do so... However, I understand that using an HTML element as the key to a dictionary may not be ideal either. So what I propose is to make a dictionary with the `data-model_name` of the form element as the key. What do you think?

Reply:
It does not matter much, but personally, I would go with a Map, potentially mapping the form element itself to its info. My arguments would be:
- The DOM is the view, the data is the model (speaking in term of MVC pattern), we should not mix both.
- We should not write random properties on standard objects as they could use it / if everyone do that we'll have collisions at some point, etc.